src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4059,6 +4110,30 @@ def create_functionapp(cmd, resource_group_name, name, storage_account, plan=Non
     return functionapp
 
 
+def _validate_cpu_momory_functionapp(cpu=None, memory=None):
+    # validate either both cup and momory are provided or none is provided. throw error otherwise
+    if cpu is None and memory is None:
+        return
+
+    if cpu is not None and memory is None:
+        raise ArgumentUsageError("The --memory argument is required with --cpu. Please provide both or none.")
+
+    if cpu is None and memory is not None:
+        raise ArgumentUsageError("The --cpu argument is required with --memory. Please provide both or none.")
```

Comment:
Nit: This might have been simplified with just:
```suggestion
    if (cpu is not None) != (memory is not None):
        raise ArgumentUsageError("The --memory argument is required with --cpu. Please provide both or none.")
```

Reply:
I wanted to keep targeted message.  I would like to keep the way it is if it is ok with you.